### DISCUSSION TITLE
Compiling .po files with Vite what is the purpose of compiling also to .ts?

### DISCUSSION THREAD

**Discussion by ManuDoni** [lingui/js-lingui#1892] [General]
Hello,
I just finished integrating Lingui in a Typescript React project.
I set up Vite and dynamic loading following the documentation, but I don't understand what is the purpose of the compiled typescript files (`lingui compile --typescript`) if I dynamically load `.po` files like this:
```typescript
export async function dynamicActivate(locale: Locale) {
  const { messages } = await import(`../locales/${locale}/messages.po`);
  i18n.load(locale, messages);
  i18n.activate(locale);
}
```
Thank tou

**Comment by andrii-bodnar** [comment]
The `--typescript` flag allows the `compile` command to produce compiled message catalogs with TypeScript types.

Please see this PR (https://github.com/lingui/js-lingui/pull/1851) to read more about the `lingui compile --typescript`.

**Comment by timofei-iatsenko** [comment]
There is no purpose in compiling separately if you already setup a dynamic loading using Vite. Vite plugin is doing the same.

**Reply by ManuDoni** [reply]
So i don't need to execute the command `lingui compile` either when I deploy to production?

**Reply by timofei-iatsenko** [reply]
Yes, you don't need if you use loader in webpack/vite like you showed.
